### ISSUE TITLE
IPv6 problems in resolvers.conf and more

### ISSUE THREAD

**Issue by swent** [jlesage/docker-nginx-proxy-manager#58]
Hello,

I get the following error:

> [app] starting Nginx Proxy Manager...
> [nginx] starting...
> nginx: [emerg] invalid port in resolver "fd00::de39:6fff:feb8:f93e" in /etc/nginx/conf.d/include/resolvers.conf:1
> [3/25/2020] [10:59:12 PM] [Migrate ] › ℹ info Current database version: 20190227065017
> 
> [3/25/2020] [10:59:12 PM] [IP Ranges] › ℹ info Fetching IP Ranges from online services...
> [3/25/2020] [10:59:12 PM] [IP Ranges] › ℹ info Fetching https://ip-ranges.amazonaws.com/ip-ranges.json
> [nginx] starting...
> nginx: [emerg] invalid port in resolver "fd00::de39:6fff:feb8:f93e" in /etc/nginx/conf.d/include/resolvers.conf:1
> [3/25/2020] [10:59:12 PM] [IP Ranges] › ℹ info Fetching https://www.cloudflare.com/ips-v4
> [3/25/2020] [10:59:12 PM] [IP Ranges] › ℹ info Fetching https://www.cloudflare.com/ips-v6
> [3/25/2020] [10:59:12 PM] [SSL ] › ℹ info Let's Encrypt Renewal Timer initialized
> [3/25/2020] [10:59:12 PM] [SSL ] › ℹ info Renewing SSL certs close to expiry...
> [3/25/2020] [10:59:12 PM] [IP Ranges] › ℹ info IP Ranges Renewal Timer initialized
> Uncaught Error: listen EADDRINUSE: address already in use :::3000
> 
> 
> FROM
> Server.emit (events.js:167:7)
> emitErrorNT (net.js:1317:8)
> process._tickCallback (internal/process/next_tick.js:63:19)
> 2020-03-25 22:59:12 63 [Warning] Aborted connection 63 to db: 'nginxproxymanager' user: 'nginxproxymanager' host: 'localhost' (Got an error reading communication packets)

The first problem is easy:
resolvers.conf looks like this:
`resolver 192.168.37.1 fd00::de39:6fff:feb8:f93e ;`
but to my understanding should look like this:
`resolver 192.168.37.1 [fd00::de39:6fff:feb8:f93e] ;`

This will fix the "invalid port in resolver" error.
I think the `resolver.conf` is generated in `/etc/cont-init.d/nginx-proxy-manager.sh:147`
`echo resolver $(awk 'BEGIN{ORS=" "} $1=="nameserver" {print $2}' /etc/resolv.conf) ";" > /config/nginx/resolvers.conf`

Second problem is `EADDRINUSE` error.
I don't know where it originates from but `:::3000` seems to be another IPv6 related syntax problem.
I think it should be `[::]:3000` instead.

I just don't know why the second error appeared after my last container update, I already had it running on IPv6 after hot-fixing the first problem myself directly on the resolvers.conf.

Thank you for this great container/app!

**Comment by swent** [comment]
I just noticed that the second error is no syntax problem, I really do have port 3000 in use on my machine.
But the problem is that it would be a lot of work (> 2 days) to move / change everything connected to my database running at that port.
Could you please map the backend app port via container variable so we can change it we want to?

**Comment by jlesage** [comment]
Could you provide the content of your `/etc/resolv.conf` ?

For the port, if you use the "bridge" network mode, you can map the port of the container to any port on your host.

**Comment by swent** [comment]
I agree if using the `bridge` network mode, but while my unraid OS has IPv6 connectivity, I was unable to get it to work for containers in bridge mode.
Thats why this might not be an option for everyone.

Content of `/etc/resolv.conf`

> /tmp # cat /etc/resolv.conf
> # Generated by dhcpcd from br0.dhcp, br0.dhcp6, br0.ra
> # /etc/resolv.conf.head can replace this line
> domain fritz.box
> nameserver 192.168.37.1
> nameserver fd00::de39:6fff:feb8:f93e
> # /etc/resolv.conf.tail can replace this line

Thanks for the help

**Comment by jlesage** [comment]
Issue with `resolvers.conf` is now fixed.
